### PR TITLE
Add roadmap document

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

As an open source project we want to be transparent with the initiatives we have planned to give users an outlook on what is ahead of the project and in which direction it is steering.

I would like to invite all @webdriverio/technical-steering-committee members to a meeting discussing the roadmap items put into the "consideration" to create a list of projects we should tackle in Q3. I am pretty flexible regarding times so please post your availability in this thread so we can schedule for a time that suits all.

### CODE REVIEW

**Comment by WillBrock** [comment]
I should be able to do any day between 7:00am and 11:00pm eastern time. I think that is 1:00pm and 5:00am Berlin time.

**Comment by mgrybyk** [comment]
I don't have much time this summer, should be available between 10:00 till 18:00 Berlin time. Days of week: Tuesday, Wednesday, Thursday. Not sure about other days.

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+| Project | Description |
+|---------|-------------|
+|TBD|TBD|
+
+## Projects Under Consideration (unplanned and unprioritized)
+
+| Project | Description |
+|---------|-------------|
+|Jest Framework Support|Jest has become one of the most popular unit test framework in the JS ecosystem. Even though most of their features are not helpful in the e2e space there are some things (e.g. snapshot testing) that could help people writing better e2e tests. There has been some initial work done on this.|
+|Auto Driver Setup|One of the most challenging problems when working with WebDriver is the setup of the driver. Even though WebdriverIO provides services around this that help to get up and running easier, it is still difficult to set up for someone who has no idea about how everything is connected. To simplify this it would be desire-able if WebdriverIO handles downloading and running drivers for all environments.|
```
Comment (mgrybyk) [inline ROADMAP.md]: `desire-able` -> `desirable`?

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+|Better Debugging Capabilities|There are already a [handful options](https://webdriver.io/docs/debugging.html) to debug test code with WebdriverIO. However it is still not straight forward to use native Node.js debugging capabilities which requires special handling of workers and sub processes. Goal has to be that developers can use their IDEs to set breakpoints to debug.|
```
Comment (mgrybyk) [inline ROADMAP.md]: Isn't it possible to debug tests and webdriverIO itself right now? I can set breakpoints in my VSCode

**Comment by mgrybyk** [comment]
One more possible project:

Not sure how to name this properly, I'd like to wrap $$() to support refetch, filtering, querying multiple items.
$$('').waitForLength(3)
$$('').waitForDisplayed()
$$('').getText() // ['text1', 'text2', 'text3']

**Comment by mgrybyk** [comment]
We can also ask people to vote for a project. The one with biggest amount of votes might be suggest as next one to be implemented

**Comment by christian-bromann** [comment]
> One more possible project:

@mgrybyk can you create a feature request with a detailed description?

> We can also ask people to vote for a project. The one with biggest amount of votes might be suggest as next one to be implemented

Every project should be outlined in an issue so that people can vote for it. We could label it as `Roadmap Item` or something like that and reach out to the Gitter/Twitter community to ask for participation.

**Comment by mgrybyk** [comment]
Yes, I'll create soon, also want to create FR for requests mocking. Similar to what Cypress have

**Comment by klamping** [comment]
I'm available most days between 10am-6pm CST, and could also sneak in a weekend meeting if need be.

**Comment by klamping** [comment]
Would it make more sense to use https://github.com/webdriverio/webdriverio/projects for hosting these TODO items? That way it's a bit easier to maintain, and comes with integrations into PRs/Branches and such

**Comment by mgrybyk** [comment]
@christian-bromann 
#4079 - mock requests
#4080 - refetch $$

**Comment by christian-bromann** [comment]
Here are some meeting suggestions, please vote by emoji:

🚀 Tue 06/24/2019 - 4pm CEST / 10am ET
❤️  Wed 06/25/2019 - 4pm CEST / 10am ET
🎉 Thu 06/26/2019 - 4pm CEST / 10am ET

👎 one of these days but earlier
👍 one of these days but later
😕 nothing really fits (please suggest a better time)

**Comment by christian-bromann** [comment]
Ok, so most of the people voted for Thu 06/26/2019 - 4pm CEST / 10am ET, so Thursday it is. Sending out invites.

**Comment by christian-bromann** [comment]
Anyone having issues joining?

**Pull request merged**